### PR TITLE
Fix amp-list-load-more doesn't shrink issue. 

### PR DESCRIPTION
**Pull request by cathyxz**
~Dependent on https://github.com/ampproject/amphtml/pull/20157.~ Merged and rebased.

### Bug
The bug is when we change between srcs in `<amp-list load-more>`, we can get into a state where the height of the `<amp-list>` is too long for its contents because amp-list doesn't shrink (https://github.com/ampproject/amphtml/issues/19939), i.e. the bottom of the amp-list is much lower than the end of the list contents. In this case, load-more will not trigger until we scroll to 3 viewports away from the bottom of the actual `<amp-list>`, which means scrolling past a lot of nothingness. 

### Solution
Refactored the `amp-list-load-more` feature to trigger `loadMoreCallback` on `viewport.onScroll` instead of using position observer. 

Basically, this bug is because the position observer is registered on the container. The solution is to measure the position of the load-more button instead. This cannot be done with position observer because position observer does not report the dimensions of elements outside of the viewport (i.e. 3 viewports away). Thus, we use a `viewport.onScroll` trigger instead and manually measure the dimensions of the viewport and the load-more-button.

### CODE REVIEW

**Comment by cathyxz** [comment]
@cvializ I sync-ed briefly with @zhouyx on this per https://github.com/ampproject/amphtml/issues/20202. This is an explicit decision due to privacy reasons, since position observer is used for more than just runtime / extension code callbacks. The TLDR is also that position observer is more expensive than necessary since it tries to measure everything on every animation frame. This approach using an async measure function should be a lot less expensive as well.

**Comment by cvializ** [comment]
Cool, thanks for helping me understand that 👍
